tests/test.hs

```diff
@@ -432,17 +435,37 @@ selfTests
 
 testGroupsWithLibs :: String -> DependentTests -> [TestTree]
 testGroupsWithLibs name (DependentTests libTests nonlibTests) =
-  [ testGroup (name <> "Lib") libTests
-  , after AllFinish (name <> "Lib") $ testGroup name nonlibTests
+  [ testGroup (name <> "-libs") libTests
+  , after AllFinish (name <> "-libs") $ testGroup name nonlibTests
   ]
 
+-- | Creates a [TestTree] that runs without parallelism
+testSequentially :: String -> SequentialTests -> [TestTree]
+testSequentially name (SequentialTests indexedTests) =
+  let grouped = (\(t, n) -> (testGroup (name <> show n) [t], n)) <$> indexedTests
+      -- turns i.e. [a, b, c, d] into [(a, b), (b, c), (c, d)]
+      pairs = zip <*> tail $ grouped
```

Comment:
Sorry, I always use this form because of https://stackoverflow.com/questions/3774247/what-do-we-call-this-new-higher-order-function/3775293#3775293.  I will change it!